README.md

```diff
@@ -1,406 +1,75 @@
 
-```csharp
-options.Events.OnTokenValidated = async context =>
-{
-    // This check is required to ensure that the Web API only accepts tokens from tenants where it has been consented and provisioned.
-    if (!context.Principal.Claims.Any(x => x.Type == ClaimConstants.Scope)
-    && !context.Principal.Claims.Any(y => y.Type == ClaimConstants.Scp)
-    && !context.Principal.Claims.Any(y => y.Type == ClaimConstants.Roles))
-    {
-         throw new UnauthorizedAccessException("Neither scope or roles claim was found in the bearer token.");
-    }
+We use [Stack Overflow](http://stackoverflow.com/questions/) with the community to provide support, using the tags `web app`, `web api`, `asp.net core`. We highly recommend you ask your questions on Stack Overflow first and browse existing issues to see if someone has asked your question before.
 
-    await Task.FromResult(0);
-};
-```
+To provide a recommendation, visit our [User Voice page](https://feedback.azure.com/forums/169401-azure-active-directory).
 
-Say you want to augment the current `ClaimsPrincipal` by adding claims to it, and you have to do it on `OnTokenValidated `. However, you don't want to lose the `UnauthorizedAccessException` check existing in the event. To do so, in your `Startup.cs`, you'd have:
+## Contribute
 
-```csharp
-services.AddProtectedWebApi(Configuration);
-services.Configure<JwtBearerOptions>(AzureADDefaults.JwtBearerAuthenticationScheme, options =>
-{
-  var existingOnTokenValidatedHandler = options.Events.OnTokenValidated ;
-  options.Events.OnTokenValidated = async context =>
-  {
-       await existingOnTokenValidatedHandler(context);
-      // your code to add extra claims that will be executed after the current event implementation.
-  }
-}
+We enthusiastically welcome contributions and feedback. You can clone the repo and start contributing now. Read our [Contribution Guide](contributing.md) for more information.
 
-```
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
 
-## Learn more about the library
+## Security Library
 
-You can learn more about the tokens by looking at the following articles in MSAL.NET's conceptual documentation:
+This library controls how users sign-in and access services. We recommend you always take the latest version of our library in your app when possible. We use [semantic versioning](http://semver.org) so you can control the risk associated with updating your app. As an example, always downloading the latest minor version number (e.g. x.*y*.x) ensures you get the latest security and feature enhancements but our API surface remains the same. You can always see the latest version and release notes under the Releases tab of GitHub.

```

Comment:
`feature enhancements but our API` > `feature enhancements, but our API`. (Microsoft Word says there should be a comma.)